### PR TITLE
Resource deallocation

### PR DESCRIPTION
**Pull request by fredoboulo**
Here is a sample test as well as the revert of the commit for the closing of the sockets.
Please note that the unit test does not end with Test so it will not be running in automatic testing, as this kind of stress test may need to run for a long period of time.

This exact same test can be used on my machine to detect resource deallocation problems in sockets as well as in polling

### CODE REVIEW

File: src/main/java/zmq/Ctx.java [hunk 1]
```diff
@@ -149,6 +149,8 @@ private void destroy() throws IOException
         termMailbox.close();
 
         tag = 0xdeadbeef;
+
+        zmq.ZMQ.closePollSelector();
```
Comment (daveyarwood) [inline src/main/java/zmq/Ctx.java]: I'm almost 2 years late to this conversation, but is there a reason that the PollSelector needs to be ThreadLocal? One thing I've read about ThreadLocal is that it makes it easy to introduce a memory leak. I think we might be seeing an example of this with the "Too many files open" problems.

File: src/main/java/zmq/Ctx.java [hunk 2]
```diff
@@ -344,6 +346,7 @@ public void destroySocket(SocketBase socket)
         try {
             int tid = socket.getTid();
             emptySlots.add(tid);
+            slots[tid].close();
```
Comment (daveyarwood) [inline src/main/java/zmq/Ctx.java]: Seems like we don't want to revert this fix. I would advocate for removing this commit.

File: src/main/java/zmq/Ctx.java (hunk 1 above)
Comment (fredoboulo) [inline src/main/java/zmq/Ctx.java]: I'm also late...
I would tend to give it an historical reason, @miniway could tell us more about that.
The tests I made on this library advocate against the use of ThreadLocal and finalize() to release the selector resource. This was also one big motivation to create the ZProxy class: having a better control on the proxy operations and suppress this memory leak for my own uses.

**Comment by daveyarwood** [comment]
Superceded by #386.

**Pull request closed without merging**